src/engine/InputQuery.cpp

```diff
@@ -867,6 +869,100 @@ bool InputQuery::constructReluLayer( NLR::NetworkLevelReasoner *nlr,
     return true;
 }
 
+bool InputQuery::constructLeakyReluLayer( NLR::NetworkLevelReasoner *nlr,
+                                          Map<unsigned, unsigned> &handledVariableToLayer,
+                                          unsigned newLayerIndex )
+{
+    INPUT_QUERY_LOG( "Attempting to construct LeakyReLULayer..." );
+    struct NeuronInformation
+    {
+    public:
+
+        NeuronInformation( unsigned variable, unsigned neuron, unsigned sourceVariable )
+            : _variable( variable )
+            , _neuron( neuron )
+            , _sourceVariable( sourceVariable )
+        {
+        }
+
+        unsigned _variable;
+        unsigned _neuron;
+        unsigned _sourceVariable;
+    };
+
+    List<NeuronInformation> newNeurons;
+
+    // Look for ReLUs where all b variables have already been handled
+    const List<PiecewiseLinearConstraint *> &plConstraints =
+        getPiecewiseLinearConstraints();
+
+    double alpha = 0;
+    for ( const auto &plc : plConstraints )
+    {
+        // Only consider ReLUs
+        if ( plc->getType() != LEAKY_RELU )
+            continue;
+
+        const LeakyReluConstraint *relu = (const LeakyReluConstraint *)plc;
+
+        // Has the b variable been handled?
+        unsigned b = relu->getB();
+        if ( !handledVariableToLayer.exists( b ) )
+            continue;
+
+        // If the f variable has also been handled, ignore this constraint
+        unsigned f = relu->getF();
+        if ( handledVariableToLayer.exists( f ) )
+            continue;
+
+        // B has been handled, f hasn't. Add f
+        newNeurons.append( NeuronInformation( f, newNeurons.size(), b ) );
+        nlr->addConstraintInTopologicalOrder( plc );
+        ASSERT( relu->getSlope() > 0 );
+        ASSERT( alpha == 0 || alpha == relu->getSlope() );
+        alpha = relu->getSlope();
+    }
+
+    // No neurons found for the new layer
+    if ( newNeurons.empty() )
+    {
+        INPUT_QUERY_LOG( "\tFailed!" );
+        return false;
+    }
+
+    nlr->addLayer( newLayerIndex, NLR::Layer::LEAKY_RELU, newNeurons.size() );
+
+    NLR::Layer *layer = nlr->getLayer( newLayerIndex );
+    layer->setAlpha( alpha );
```

Comment:
Is Alpha the used term for a leaky relu slope? If so, then okay, but otherwise lets change to slope

Reply:
Yes, this is the common term.